addons/mail/static/src/discuss/core/common/thread_model_patch.js

```diff
@@ -35,7 +36,90 @@ const threadPatch = {
                 return this.otherTypingMembers.length > 0;
             },
         });
+        this.hasSeenFeature = Record.attr(false, {
+            /** @this {import("models").Thread} */
+            compute() {
+                return this.store.channel_types_with_seen_infos.includes(this.channel_type);
+            },
+        });
+        this.firstUnreadMessage = Record.one("mail.message", {
+            /** @this {import("models").Thread} */
+            compute() {
+                if (!this.selfMember) {
+                    return null;
+                }
+                const messages = this.nonEmptyMessages;
+                const separator = this.selfMember.localNewMessageSeparator;
+                if (separator === 0 && !this.loadOlder) {
+                    return messages[0];
+                }
+                if (!separator || messages.length === 0 || messages.at(-1).id < separator) {
+                    return null;
+                }
+                // try to find a perfect match according to the member's separator
+                let message = this.store["mail.message"].get({ id: separator });
+                if (!message || this.notEq(message.thread) || message.isEmpty) {
+                    message = nearestGreaterThanOrEqual(messages, separator, (msg) => msg.id);
+                }
+                return message;
+            },
+            inverse: "threadAsFirstUnread",
+        });
         this.invitedMembers = Record.many("discuss.channel.member");
+        this.last_interest_dt = Record.attr(undefined, { type: "datetime" });
+        /** @type {luxon.DateTime} */
+        this.lastInterestDt = Record.attr(undefined, {
+            type: "datetime",
+            /** @this {import("models").Thread} */
+            compute() {
+                const selfMemberLastInterestDt = this.selfMember?.last_interest_dt;
+                const lastInterestDt = this.last_interest_dt;
+                return compareDatetime(selfMemberLastInterestDt, lastInterestDt) > 0
+                    ? selfMemberLastInterestDt
+                    : lastInterestDt;
+            },
+        });
+        this.lastMessageSeenByAllId = Record.attr(undefined, {
+            compute() {
+                if (!this.hasSeenFeature) {
+                    return;
+                }
+                const otherMembers = this.channel_member_ids.filter((member) =>
+                    member.persona.notEq(this.store.self)
+                );
+                if (otherMembers.length === 0) {
+                    return;
+                }
+                const otherLastSeenMessageIds = otherMembers
+                    .filter((member) => member.seen_message_id)
+                    .map((member) => member.seen_message_id.id);
+                if (otherLastSeenMessageIds.length === 0) {
+                    return;
+                }
+                return Math.min(...otherLastSeenMessageIds);
```

Comment:
Could be simplified as well.

```suggestion
                return Math.min(otherLastSeenMessageIds);
```